### ISSUE TITLE
Rule proposal: `prefer-script-setup`

### ISSUE THREAD

**Issue by cexbrayat** [vuejs/eslint-plugin-vue#1624]
**Please describe what the rule should do:**

As `script setup` is slowing becoming the recommended way to declare a component, it would be nice to have a rule that checks the components of a project and warn the author if a component does not use `script setup`

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<script lang="ts">
import { defineComponent } from 'vue';
export default defineComponent({
 //...
});
</script>
```

This should not warn:
```vue
<script setup lang="ts">
</script>
```

**Comment by ota-meshi** [comment]
Thank you for the rule proposal! Sounds good to me!
However, I think some users may prefer the plain Composition API or Objects API.
Do you think the rule should have options to work for these users as well?

**Comment by cexbrayat** [comment]
That's a good point. Maybe it's a good idea to have options for this. As Composition API seems to be what should be favored, that could be the default?
